c/src/lib/libbsp/arm/shared/stm32fxxxx/include/hal-spi.h

```diff
@@ -0,0 +1,133 @@
+/**
+ * @file hal-i2c.h
+ *
+ * @ingroup i2c
+ *
+ * @brief Public I2C driver datatypes
+ *
+ */
+
+/*
+ * Copyright (c) 2015 Vecna Technologies, Inc.
+ *
+ * The license and distribution terms for this file may be
+ * found in the file LICENSE in this distribution or at
+ * http://www.rtems.com/license/LICENSE.
+ */
+
+#ifndef STM32_SPI_H
+#define STM32_SPI_H
+
+#include <stdint.h>
+#include <stdbool.h>
+
+#include <dev/spi/spi.h>
+
+
+/************************ Defines ******************************/
+#define STRINGIFY(x)  #x
+
+
+#define _PIN(x)       GPIO_PIN_##x
+#define PIN(x)        STRINGIFY(_PIN(x))
+
+#define _GET_SPI_VEC(x)                SPI##x_IRQn
+#define GET_SPI_VEC(__INSTANCE__)      STRINGIFY(_GET_SPI_VEC(__INSTANCE__))
+
```

Comment:
since there is some macro magic here I think macro documentation and comments would be nice... Intended use case and the like.